framework/src/org/checkerframework/framework/type/TypeHierarchy.java

```diff
@@ -8,13 +10,58 @@
  */
 public interface TypeHierarchy {
 
+
     /**
-     * @return true if subtype {@literal <:} supertype for all qualifier hierarchies in the type system
+     * Returns true if {@code subtype} is a subtype of or convertible to {@code supertype}
+     * for all hierarchies present.  If the underlying Java type of {@code subtype} must be a subtype of
+     * or convertible to the underlying Java type of {@code supertype}, then the behavior of this method is undefined.
+     *
+     * Ideally, types that require conversions would be converted before isSubtype is called, but instead, isSubtype
+     * performs some of these conversions.
+     *
+     * JLS 5.1 specifies 13 categories of conversions.  4 categories are converted in isSubtype:
```

Comment:
I wouldn't start a sentence with the literal 4. Same with `1` below, where also the indentation is off.